Doc/library/multiprocessing.rst

```diff
@@ -1168,6 +1168,66 @@ For example:
     the data in the pipe is likely to become corrupted, because it may become
     impossible to be sure where the message boundaries lie.
 
+Custom Reduction
+~~~~~~~~~~~~~~~~
+
+.. currentmodule:: multiprocessing
+
+Several primitives of the :mod:`multiprocessing` module such as
+:class:`multiprocessing.Queue`, :class:`multiprocessing.connection.Listener` or
+:class:`multiprocessing.connection.Server` need to serialize and deserialize Python
+objects to communicate between processes. Sometimes it is useful to control what
+serialization is to be used for the transport of data in order to support communication
+with different versions of Python, use more performant third party tools or custom
+strategies.
+
+For this purpose a set of hooks is available to provide alternate implementations of
+the reduction mechanism:
+
+.. currentmodule:: multiprocessing.reduction
+
+.. class:: AbstractReducer()
+
+   Abstract base class that can be implemented in order to replace the standard
+   reduction mechanism used in multiprocessing
+
+   .. function:: get_pickler():
+
+      This method must return an instance of a subclass of :class:`pickler.Pickler`
+      to be used by the multiprocessing reducer mechanism.
+
+.. currentmodule:: multiprocessing
+
+.. method:: set_reducer(reduction)
+
+   Sets a reduction class to be used for serialization and deserialization by the module
+   primitive internals. **reduction** must be an instance of a subclass of
+   :class:`multiprocessing.reduction.AbstractReducer`.
+
+.. method:: get_reducer()
+
+   Gets the current reduction class in use by the module's primitive internals.
+
+For example, substituting the reducer class to use the :mod:`pickle` protocol
+version 2 to be able to communicate with a Python 2.x programs.::
+
+   import multiprocessing
+   from multiprocessing.reduction import AbstractReducer, ForkingPickler
+
+   class ForkingPicklerProtocol2(ForkingPickler):
+       @classmethod
+       def dumps(cls, obj, pickle_protocol=2):
```

Comment:
By the way, is it the `dumps` method which should be overriden or rather the `dump` method?

Reply:
This question is still relevant... apparently it's both ;-)

Reply:
Yep, is both at this moment, would you prefer that the `AbstractReducer` class implements dumps on top of dump or to force the user to implement both?

We could also introduce a new abstract class instead of demanding "a subclass of Pickler".

Reply:
In my last commit (7ee45e1) I have abstracted what's required for the reducer class, basically reducing the methods to be implemented to `dumps` and `loads`.